src/core/redeem.rs

```diff
@@ -171,7 +171,14 @@ impl<J: Jet> RedeemNode<J> {
         let witness = WitnessDecoder::new(bits)?;
         let program = commit.finalize(witness, false)?;
 
-        if sharing::check_maximal_sharing((*program).post_order_iter::<FullSharing>()) {
+        let iter = (&*program)
+            .post_order_iter::<InternalSharing>()
+            .map(|node| node.imr);
+        let fully_shared_iter = (&*program)
```

Comment:
`&* program` can be `program.as_ref()`

Reply:
Pushed fix for this (and one other instance in decode.rs where I did the same thing).

Will add a sharing unit test in a fresh commit.